kratos/mpi/tests/sources/test_mpi_assign_unique_model_part_collection_tag_utility.cpp

```diff
@@ -0,0 +1,117 @@
+//    |  /           |
+//    ' /   _| _` | _|  _ \   _|
+//    . \  |   (   | |   (   |\_ `
+//   _|\_\_|  \_,_|\_|\__/ __/
+//                   Multi-Physics
+//
+//  License:   BSD License
+//  Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz, Marc Nunez
+//
+
+// System includes
+#include <unordered_set>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "testing/testing.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_filesystem.h"
+#include "utilities/cpp_tests_utilities.h"
+#include "mpi/includes/mpi_communicator.h"
+
+/* Utilities */
+#include "utilities/assign_unique_model_part_collection_tag_utility.h"
+
+namespace Kratos
+{
+    namespace Testing
+    {
+        typedef Node<3> NodeType;
+        typedef std::size_t IndexSize;
+        typedef std::unordered_map<IndexSize,IndexSize> IndexIndexMapType;
+        typedef std::unordered_map<IndexSize,std::vector<std::string>> IndexStringMapType;
+
+        /**
+        * Checks the correct work of the sub modelparts list utility
+        */
+        KRATOS_TEST_CASE_IN_SUITE(AssignMPIUniqueModelPartCollectionTagUtility, KratosMPICoreFastSuite)
+        {
+            // Creating the reference model part and the relative submodelparts non alphabetically ordered
+            Model current_model;
+            ModelPart& r_model_part = current_model.CreateModelPart("Main");
+            ModelPart& r_sub_modelpart_1 = r_model_part.CreateSubModelPart("BSubModelPart1");
+            ModelPart& r_sub_modelpart_2 = r_model_part.CreateSubModelPart("ASubModelPart2");
+            ModelPart& r_sub_modelpart_3 = r_model_part.CreateSubModelPart("ZSubModelPart3");
+            ModelPart& r_sub_modelpart_4 = r_model_part.CreateSubModelPart("YSubModelPart4");
+
+            Communicator::Pointer pnew_comm = Kratos::make_shared< MPICommunicator >(&r_model_part.GetNodalSolutionStepVariablesList(), DataCommunicator::GetDefault());
```

Comment:
I think this doesn't set it correctly for the SubModelParts (thats what he ParallelFillCommunicator does, but for that one you need to properly specify the partition index etc)
=> if it works like this is ok for me, but be careful, this has the potential for deadlocks/bugs

Reply:
For this test I think it is fine. As you say the problem is that I am using the test model_part created manually, so I cant use the parallel fill comm. 

To avoid this I could read a proper mdpa, use the partition utility and call the ParallelFill, but for the purpose of the test I think it is fine as it is.

Reply:
you could still do it manually, but it is more complicated:
https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/mpi/tests/sources/test_mpi_communicator.cpp#L30-L66

=> for such cases it would be nice to be able to partition an existing ModelPart
but ... future